designs/2019-core-options/README.md

```diff
@@ -0,0 +1,139 @@
+
+<table><td>
+<b>💠Implementation</b>:
+<ol>
+<li><code>Linter</code> and <code>CLIEngine</code> have <code>options.reportUnusedDisableDirectives</code>. This RFC enhances these options to accept <code>"off"</code>, <code>"warn"</code>, and <code>"error"</code>. Existing <code>false</code> is the same as <code>"off"</code> and existing <code>true</code> is the same as <code>"error"</code>.</li>
+<li>In <a href="https://github.com/eslint/eslint/blob/af81cb3ecc5e6bf43a6a2d8f326103350513a1b8/lib/linter.js#L859"><code>Linter#_verifyWithoutProcessors</code> method</a>, the linter checks both <code>providedConfig</code> and <code>filenameOrOptions</code> to determine <code>reportUnusedDisableDirectives</code> option. The <code>filenameOrOptions.reportUnusedDisableDirectives</code> precedences <code>providedConfig.reportUnusedDisableDirectives</code>.</li>
+</ol>
+</td></table>
+
+### § verifyOnRecoverableParsingErrors
+
+That value can be a boolean value. Default is `false`.
+
+If `true` then it runs rules even if recoverable errors existed. Then it shows both results.
+
+<table><td>
+<b>💠Implementation</b>:
+<p>In <a href="https://github.com/eslint/eslint/blob/af81cb3ecc5e6bf43a6a2d8f326103350513a1b8/lib/linter.js#L859"><code>Linter#_verifyWithoutProcessors</code> method</a>, the linter checks both <code>providedConfig</code> and <code>filenameOrOptions</code> to determine <code>verifyOnRecoverableParsingErrors</code> option. The <code>filenameOrOptions.verifyOnRecoverableParsingErrors</code> precedences <code>providedConfig.verifyOnRecoverableParsingErrors</code>.</p>
+</td></table>
+
+### § ignorePatterns
+
+That value can be an array of strings. Default is an empty array.
+
+This is very similar to `.eslintignore` file. Each value is a file pattern as same as each line of `.eslintignore` file. ESLint compares the path to source code files and the file pattern then it ignores the file if it was matched. The path to source code files is addressed as relative to the entry config file, as same as `files`/`excludedFiles` properties.
+
+ESLint concatenates all ignore patterns from all of `.eslintignore`, `--ignore-path`, `--ignore-pattern`, and `ignorePatterns`. If there are multiple `ignorePatterns` in a `ConfigArray`, all of them are concatenated. The order is:
+
+1. The default ignoring. (I.e. `.*`, `node_modules/*`, and `bower_components/*`)
+1. `ignorePatterns` in the appearance order in the config array.
+1. `--ignore-path` or `.eslintignore`.
+1. `--ignore-pattern`
+
+Negative patterns mean unignoring. For example, `!.*.js` makes ESLint checking JavaScript files which start with `.`. Negative patterns are used to override parent settings.
+Also, negative patterns is worthful for shareable configs of some platforms. For example, the config of VuePress can provide the configuration that unignores `.vuepress` directory.
```

Comment:
I argue unignoring files is not a good pratice. maybe we can reconsider not ignoring `.*`

Reply:
It's another matter. :)

This just provides a way to configure ignoring patterns. It's the way to override the patterns in the extended configs. Also, it can override the default config.

If you want to change the default pattern, you can write an RFC.